lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -2300,4 +2271,53 @@ public function walkResultVariable($resultVariable)
 
         return $resultAlias;
     }
+
+    /**
+     * @param ClassMetadataInfo $discrClass
+     * @param AST\InstanceOfExpression $instanceOfExpr
+     * @return string The list in parentheses of valid child discriminators from the given class
+     * @throws QueryException
+     */
+    private function getChildDiscriminatorsFromClassMetadata(ClassMetadataInfo $discrClass, AST\InstanceOfExpression $instanceOfExpr)
+    {
+        $knownSubclasses = array_flip($discrClass->subClasses);
+        $sqlParameterList = [];
+        $discriminators = [];
+        foreach ($instanceOfExpr->value as $parameter) {
+            if ($parameter instanceof AST\InputParameter) {
+                $this->rsm->addMetadataParameterMapping($parameter->name, 'discriminatorValue');
+
+                $sqlParameterList[] = $this->walkInputParameter($parameter);
+
+                continue;
+            }
+
+            // Trim first backslash
+            $parameter = ltrim($parameter, '\\');
```

Comment:
Right! I was also already doing the `getClassMetadata` a few lines after, I'll just move that up!